core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -573,19 +573,28 @@ public boolean hasNext() {
 
     @Override
     public void loadNext() throws IOException {
-      synchronized (this) {
```

Comment:
is it possible lastPage != null on line 550 above when it is set? https://github.com/apache/spark/pull/24269/files#diff-027299fb14327ddcaba457f81ecff32cR550

Reply:
It's possible but doesn't matter. It's kind of we delay the page releasing a little bit.

Reply:
note that both the 2 methods are inside `synchronized (this)`